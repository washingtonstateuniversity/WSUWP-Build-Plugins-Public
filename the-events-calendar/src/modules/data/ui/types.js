/**
 * Internal dependencies
 */
import { PREFIX_EVENTS_STORE } from '@moderntribe/events/data/utils';

export const SET_DASHBOARD_DATE_TIME = `${ PREFIX_EVENTS_STORE }/SET_DASHBOARD_DATE_TIME`;
export const TOGGLE_DASHBOARD_DATE_TIME = `${ PREFIX_EVENTS_STORE }/TOGGLE_DASHBOARD_DATE_TIME`;

export const SET_DASHBOARD_PRICE = `${ PREFIX_EVENTS_STORE }/SET_DASHBOARD_PRICE`;

export const SET_VISIBLE_MONTH = `${ PREFIX_EVENTS_STORE }/SET_VISIBLE_MONTH`;
