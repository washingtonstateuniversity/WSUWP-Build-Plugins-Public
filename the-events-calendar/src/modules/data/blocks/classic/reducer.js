/**
 * Internal dependencies
 */
import * as types from './types';

export const DEFAULT_STATE = {
	detailsTitle: '',
	organizerTitle: '',
};

export default ( state = DEFAULT_STATE, action ) => {
	switch ( action.type ) {
		case types.SET_CLASSIC_DETAILS_TITLE:
			return {
				...state,
				detailsTitle: action.payload.title,
			};

		case types.SET_CLASSIC_ORGANIZERS_TITLE:
			return {
				...state,
				organizerTitle: action.payload.title,
			};

		default:
			return state;
	}
};
