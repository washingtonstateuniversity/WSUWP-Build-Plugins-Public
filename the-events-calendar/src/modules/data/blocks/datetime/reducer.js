/**
 * External dependencies
 */
import { __ } from '@wordpress/i18n';
import moment from 'moment/moment';

/**
 * Internal dependencies
 */
import {
	globals,
	date,
	moment as momentUtil,
	time,
} from '@moderntribe/common/utils';
import * as types from './types';

const defaultStartTime = globals.defaultTimes().start ? globals.defaultTimes().start : '08:00:00';
const defaultEndTime = globals.defaultTimes().end ? globals.defaultTimes().end : '17:00:00';
const defaultStartTimeSeconds = time.toSeconds( defaultStartTime, time.TIME_FORMAT_HH_MM_SS );
const defaultEndTimeSeconds = time.toSeconds( defaultEndTime, time.TIME_FORMAT_HH_MM_SS );

export const defaultStartMoment = moment().startOf( 'day' ).seconds( defaultStartTimeSeconds );
export const defaultEndMoment = moment().startOf( 'day' ).seconds( defaultEndTimeSeconds );

export const DEFAULT_STATE = {
	start: momentUtil.toDateTime( defaultStartMoment ),
	end: momentUtil.toDateTime( defaultEndMoment ),
	startTimeInput: momentUtil.toTime( defaultStartMoment ),
	endTimeInput: momentUtil.toTime( defaultEndMoment ),
	naturalLanguage: '',
	dateTimeSeparator: globals.settings().dateTimeSeparator
		? globals.settings().dateTimeSeparator
		: __( '@', 'the-events-calendar' ),
	timeRangeSeparator: globals.settings().timeRangeSeparator
		? globals.settings().timeRangeSeparator
		: __( '-', 'the-events-calendar' ),
	allDay: false,
	multiDay: false,
	timeZone: date.FORMATS.TIMEZONE.string,
	timeZoneLabel: date.FORMATS.TIMEZONE.string,
	showTimeZone: false,
	showDateInput: false,
	isEditable: true,
};

export default ( state = DEFAULT_STATE, action ) => {
	switch ( action.type ) {
		case types.SET_DATETIME_BLOCK_EDITABLE_STATE:
			return {
				...state,
				isEditable: action.payload.isEditable,
			};
		case types.SET_START_DATE_TIME:
			return {
				...state,
				start: action.payload.start,
			};
		case types.SET_END_DATE_TIME:
			return {
				...state,
				end: action.payload.end,
			};
		case types.SET_START_TIME_INPUT:
			return {
				...state,
				startTimeInput: action.payload.startTimeInput,
			};
		case types.SET_END_TIME_INPUT:
			return {
				...state,
				endTimeInput: action.payload.endTimeInput,
			};
		case types.SET_NATURAL_LANGUAGE_LABEL:
			return {
				...state,
				naturalLanguage: action.payload.label,
			};
		case types.SET_ALL_DAY:
			return {
				...state,
				allDay: action.payload.allDay,
			};
		case types.SET_MULTI_DAY:
			return {
				...state,
				multiDay: action.payload.multiDay,
			};
		case types.SET_SEPARATOR_DATE:
			return {
				...state,
				dateTimeSeparator: action.payload.separator,
			};
		case types.SET_SEPARATOR_TIME:
			return {
				...state,
				timeRangeSeparator: action.payload.separator,
			};
		case types.SET_TIME_ZONE:
			return {
				...state,
				timeZone: action.payload.timeZone,
			};
		case types.SET_TIMEZONE_LABEL:
			return {
				...state,
				timeZoneLabel: action.payload.label,
			};
		case types.SET_TIMEZONE_VISIBILITY:
			return {
				...state,
				showTimeZone: action.payload.show,
			};
		case types.SET_DATE_INPUT_VISIBILITY:
			return {
				...state,
				showDateInput: action.payload.show,
			};
		default:
			return state;
	}
};
