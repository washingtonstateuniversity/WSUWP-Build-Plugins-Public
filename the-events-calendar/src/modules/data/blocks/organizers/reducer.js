/**
 * External dependencies
 */
import { combineReducers } from 'redux';

/**
 * Internal dependencies
 */
import { classic, blocks } from './reducers';

export default combineReducers( {
	blocks,
	classic,
} );
