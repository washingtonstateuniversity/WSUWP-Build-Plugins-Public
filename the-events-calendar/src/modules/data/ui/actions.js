/**
 * Internal dependencies
 */
import * as types from './types';
import { moment as momentUtil } from '@moderntribe/common/utils';

export const toggleDashboardDateTime = () => ( {
	type: types.TOGGLE_DASHBOARD_DATE_TIME,
} );

export const openDashboardDateTime = () => ( {
	type: types.SET_DASHBOARD_DATE_TIME,
	payload: {
		open: true,
	},
} );

export const closeDashboardDateTime = () => ( {
	type: types.SET_DASHBOARD_DATE_TIME,
	payload: {
		open: false,
	},
} );

export const openDashboardPrice = () => ( {
	type: types.SET_DASHBOARD_PRICE,
	payload: {
		open: true,
	},
} );

export const closeDashboardPrice = () => ( {
	type: types.SET_DASHBOARD_PRICE,
	payload: {
		open: false,
	},
} );

export const setVisibleMonth = ( visibleMonth ) => ( {
	type: types.SET_VISIBLE_MONTH,
	payload: {
		visibleMonth,
	},
} );

export const setInitialState = ( { get } ) => ( dispatch ) => {
	const start = get( 'start' );
	if ( ! start ) {
		return;
	}

	const month = momentUtil.toMoment( momentUtil.parseFormats( start ) ).startOf( 'month' ).toDate();
	dispatch( setVisibleMonth( month ) );
};
