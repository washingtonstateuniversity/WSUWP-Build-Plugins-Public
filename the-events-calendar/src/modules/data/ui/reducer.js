/**
 * External dependencies
 */
import moment from 'moment/moment';

/**
 * Internal dependencies
 */
import * as types from './types';

export const DEFAULT_STATE = {
	dashboardDateTimeOpen: false,
	dashboardPriceOpen: false,
	visibleMonth: moment().startOf( 'month' ).toDate(),
};

export default ( state = DEFAULT_STATE, action ) => {
	switch ( action.type ) {
		case types.SET_DASHBOARD_DATE_TIME:
			return {
				...state,
				dashboardDateTimeOpen: action.payload.open,
			};
		case types.TOGGLE_DASHBOARD_DATE_TIME:
			return {
				...state,
				dashboardDateTimeOpen: ! state.dashboardDateTimeOpen,
			};
		case types.SET_DASHBOARD_PRICE:
			return {
				...state,
				dashboardPriceOpen: action.payload.open,
			};
		case types.SET_VISIBLE_MONTH:
			return {
				...state,
				visibleMonth: action.payload.visibleMonth,
			};
		default:
			return state;
	}
};
