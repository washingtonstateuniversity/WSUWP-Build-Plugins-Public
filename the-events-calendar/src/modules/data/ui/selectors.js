/**
 * External dependencies
 */
import { createSelector } from 'reselect';

export const getUI = ( state ) => state.events.ui;

export const getDashboardDateTimeOpen = createSelector(
	[ getUI ],
	( ui ) => ui.dashboardDateTimeOpen
);

export const getDashboardPriceOpen = createSelector(
	[ getUI ],
	( ui ) => ui.dashboardPriceOpen
);

export const getVisibleMonth = createSelector(
	[ getUI ],
	( ui ) => ui.visibleMonth
);
