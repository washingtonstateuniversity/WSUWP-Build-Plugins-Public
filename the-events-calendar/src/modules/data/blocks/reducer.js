/**
 * External dependencies
 */
import { combineReducers } from 'redux';

/**
 * Internal dependencies
 */
import datetime from './datetime';
import organizers from './organizers';
import price from './price';
import website from './website';
import venue from './venue';
import classic from './classic';
import sharing from './sharing';

export default combineReducers( {
	datetime,
	classic,
	venue,
	organizers,
	price,
	website,
	sharing,
} );
