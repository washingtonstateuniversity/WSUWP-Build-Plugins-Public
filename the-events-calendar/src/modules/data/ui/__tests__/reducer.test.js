/**
 * External dependencies
 */
import moment from 'moment';

/**
 * Internal dependencies
 */
import { actions } from '@moderntribe/events/data/ui';
import reducer, { DEFAULT_STATE } from '@moderntribe/events/data/ui/reducer';

describe( '[STORE] - UI reducer', () => {
	it( 'Should return the default state', () => {
		expect( reducer( undefined, {} ) ).toEqual( DEFAULT_STATE );
	} );

	it( 'Should open the Dashboard price', () => {
		expect( reducer( {}, actions.openDashboardPrice() ) ).toMatchSnapshot();
	} );

	it( 'Should Set the Dashboard date time by open it', () => {
		const current = reducer( {}, actions.openDashboardDateTime() );
		expect( current ).toMatchSnapshot();
	} );

	it( 'Should set the Dashboard date time by closing it', () => {
		const current = reducer( {}, actions.closeDashboardDateTime() );
		expect( current ).toMatchSnapshot();
	} );

	it( 'Should close the dashaboard price', () => {
		expect( reducer( {}, actions.closeDashboardPrice() ) ).toMatchSnapshot();
	} );

	it( 'Should Toggle Dashboard date time', () => {
		const current = reducer( { dashboardDateTimeOpen: false }, actions.toggleDashboardDateTime() );
		expect( current ).toMatchSnapshot();
	} );

	it( 'Should Set visibility month', () => {
		const visibleMonth = moment().startOf( 'month' ).toDate();
		const current = reducer( {}, actions.setVisibleMonth( visibleMonth ) );
		const expected = {
			visibleMonth,
		};
		expect( current ).toEqual( expected );
	} );
} );
