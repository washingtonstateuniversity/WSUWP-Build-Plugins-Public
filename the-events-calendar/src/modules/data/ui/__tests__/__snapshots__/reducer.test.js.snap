// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[STORE] - UI reducer Should Set the Dashboard date time by open it 1`] = `
Object {
  "dashboardDateTimeOpen": true,
}
`;

exports[`[STORE] - UI reducer Should Toggle Dashboard date time 1`] = `
Object {
  "dashboardDateTimeOpen": true,
}
`;

exports[`[STORE] - UI reducer Should close the dashaboard price 1`] = `
Object {
  "dashboardPriceOpen": false,
}
`;

exports[`[STORE] - UI reducer Should open the Dashboard price 1`] = `
Object {
  "dashboardPriceOpen": true,
}
`;

exports[`[STORE] - UI reducer Should set the Dashboard date time by closing it 1`] = `
Object {
  "dashboardDateTimeOpen": false,
}
`;
